server/src/graphql/mutations/createAppGithub.ts

```diff
@@ -0,0 +1,111 @@
+import { deployAppQueue } from './../../queues/deployApp';
+import { sshConnect } from './../../lib/ssh';
+import { MutationResolvers } from '../../generated/graphql';
+import { Octokit } from '@octokit/rest';
+import { prisma } from '../../prisma';
+import crypto from 'crypto';
+// import { buildAppQueue } from '../../queues/buildApp';
+import { appNameSchema, getRepoData } from '../utils';
+import { dokku } from '../../lib/dokku';
+
+export const createAppGithub: MutationResolvers['createAppGithub'] = async (
+  _,
+  { input },
+  { userId }
+) => {
+  if (!userId) {
+    throw new Error('Unauthorized');
+  }
+
+  // We make sure the name is valid to avoid security risks
+  appNameSchema.validateSync({ name: input.name });
+
+  const apps = await prisma.app.findMany({
+    where: {
+      name: input.name,
+    },
+  });
+
+  const isAppNameTaken = !!apps[0];
+
+  if (isAppNameTaken) {
+    throw new Error('App name already taken');
+  }
+
+  const octokit = new Octokit({});
+
+  let repo;
+
+  const repoData = getRepoData(input.gitRepoUrl);
+  try {
+    repo = await octokit.repos.get({
```

Comment:
here we should that the repo exist but we should also check that the branch exist for that repo otherwise it could create issues later